sqlest/src/main/scala/sqlest/ast/Merge.scala

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright 2014 JHC Systems Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sqlest.ast
+
+import sqlest.ast.syntax._
+
+case class Merge[R <: Relation](
+    into: Table,
+    using: R,
+    whenMatched: Option[MatchedOp] = None,
+    whenMatchedAnd: List[MatchedAndOp] = List(),
+    whenNotMatched: Option[NotMatchedOp] = None,
+    whenNotMatchedAnd: List[NotMatchedAndOp] = List(),
```

Comment:
yes if I make the and property optional

Reply:
Actually now that I make these changes I recall why these types exist. I want to prevent the user from adding an `and` clause when they use the merge method `whenMatched` or `whenNotMatched` and I want to ensure they do include the `and` clause when using `whenMatchedAnd` and `whenNotMatchedAnd`. I went down the route of `List[MatchedOp]` and `List[NotMatchedOp]` earlier and it was not suitable for imposing the syntax rules of the merge statement so I'd like to leave this as is unless you can advise something?